crates/bevy_ui/src/focus.rs

```diff
@@ -56,82 +56,62 @@ pub struct State {
     entities_to_reset: SmallVec<[Entity; 1]>,
 }
 
-/// The system that sets Interaction for all UI elements based on the mouse cursor activity
-#[allow(clippy::type_complexity)]
+pub type NodeQuery<'a> = (
+    Entity,
+    &'a Node,
+    &'a GlobalTransform,
+    &'a mut Interaction,
+    Option<&'a FocusPolicy>,
+    Option<&'a CalculatedClip>,
+);
+
+/// The system that sets Interaction for all UI elements based on the mouse and touch cursor
+/// activity
 pub fn ui_focus_system(
-    mut state: Local<State>,
+    state: Local<State>,
     windows: Res<Windows>,
     mouse_button_input: Res<Input<MouseButton>>,
     touches_input: Res<Touches>,
-    mut node_query: Query<(
-        Entity,
-        &Node,
-        &GlobalTransform,
-        Option<&mut Interaction>,
-        Option<&FocusPolicy>,
-        Option<&CalculatedClip>,
-    )>,
+    node_query: Query<NodeQuery>,
 ) {
-    let cursor_position = windows
-        .get_primary()
-        .and_then(|window| window.cursor_position());
-
-    // reset entities that were both clicked and released in the last frame
-    for entity in state.entities_to_reset.drain(..) {
-        if let Ok(mut interaction) = node_query.get_component_mut::<Interaction>(entity) {
-            *interaction = Interaction::None;
-        }
-    }
+    focus_ui(
```

Comment:
I don't think it's useful to have a system whose contents are a single function call: that seems like needless layering.

Reply:
This is to make it testable. Or do you have other suggestions how to mock `Windows`?

I.e. `ui_focus_system` is actually meant as a public export of `focus_ui::<Windows>`

Reply:
Ah. I think we should try to tackle that directly: I don't want to have to replicate this strategy everywhere.